### PR TITLE
Tests created

### PR DESCRIPTION
**Pull request by IParov**

### CODE REVIEW

File: src/main/java/core/basesyntax/custom/exceptions/AgeNotValidException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package core.basesyntax.custom.exceptions;
+
+public class AgeNotValidException extends RuntimeException {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/custom/exceptions/AgeNotValidException.java:3]: One Exception class is good enough.
```suggestion
public class RegistrationException extends RuntimeException {
```

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 2]
```diff
@@ -1,14 +1,39 @@
...
+import core.basesyntax.custom.exceptions.LoginNotValidException;
+import core.basesyntax.custom.exceptions.PasswordNotValidException;
+import core.basesyntax.custom.exceptions.UserNotValidException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MAX_AGE = 130;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:13]: Developer should implement the task. Developer **should not** invent the task.
Watch this video https://www.youtube.com/watch?v=Ot2eB07rjcI

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 3]
```diff
@@ -1,14 +1,39 @@
...
+    private static final int PASSWORD_SIZE = 6;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:14]:
```suggestion
    private static final int MIN_PASSWORD_SIZE = 6;
```

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 4]
```diff
@@ -1,14 +1,39 @@
...
+        if (user.getLogin() == null) {
+            throw new LoginNotValidException("Login can't be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new LoginNotValidException("Login already exist");
+        }
+        if (user.getPassword() == null) {
+            throw new PasswordNotValidException("Password can't be null");
+        }
+        if (user.getPassword().length() != PASSWORD_SIZE) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:31]: image

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 5]
```diff
@@ -1,14 +1,39 @@
...
+            throw new PasswordNotValidException("Password not valid");
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:32]: Make msg more informative

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 6]
```diff
@@ -1,14 +1,39 @@
...
+        }
+        if (user.getAge() == null || user.getAge() < MIN_AGE || user.getAge() > MAX_AGE) {
+            throw new AgeNotValidException("Not valid age");
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:35]: Make msg more informative

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:19]: empty line

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        user.setLogin(DEFAULT_LOGIN);
+        user.setPassword(DEFAULT_PASSWORD);
+        user.setAge(DEFAULT_AGE);
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        user = null;
+        assertThrows(UserNotValidException.class, () -> registrationService.register(user));
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:46]:
```suggestion
        assertThrows(UserNotValidException.class, () -> registrationService.register(null));
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(AgeNotValidException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_greatAge_notOk() {
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:62]: remove this test

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(LoginNotValidException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_sameLogin_notOk() {
+        StorageDao storageDao = new StorageDaoImpl();
+        storageDao.add(user);
+        assertThrows(LoginNotValidException.class, () -> registrationService.register(user));
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:77]: you should try to register different users with same login

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java (hunk 2 above)
Comment (IParov) [reply src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:13]: Thanks

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**